File: prob1.py
# Time Complexity : O(N)
# Space Complexity : O(1) as no extra space
# Did this code successfully run on Leetcode : Yes
# Any problem you faced while coding this : It took some time to understand the logic clearly


class Solution:

    def removeDups2(self, nums) -> int:

        slow, count = 1, 1
        for fast in range(1, len(nums)):

            if nums[fast] == nums[fast - 1]:
                count += 1
            else:
                count = 1
            
            if count < 3:
                nums[slow] = nums[fast]
                slow += 1
        return slow

File: prob1.java
// Time Complexity : O (n) 
// Space Complexity : O (1) no extra space
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : minor issues

public class prob1 {

    public static int removeDupes(int[] nums){
        if(nums == null || nums.length == 0) return 0;
        int j = 1; int count = 1;
        for(int i = 1; i < nums.length; i++){
            if(nums[i] == nums[i-1]){
                count++;
            }else{
                count = 1;
            }

            if(count < 3){
                nums[j] = nums[i];
                j++;
            }
        }
        return j;
    }

    public static void main(String[] args) {
        int [] nums = {2, 0, 1, 0, 2, 1, 0, 0, 2, 2, 1, 1, 1, 1};
        int j = removeDupes(nums);
        System.out.println(j);
        for(int i = 0; i < j; i++){
            System.out.println(nums[i]);
        }
        
    }
    
}
